coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -3759,39 +3765,190 @@ def _internal_op_tensor_inplace_fill(context, node):
+
+
+@register_torch_op
+def slice_scatter(context, node):
+    inputs = _get_inputs(context, node, min_expected=2)
+    x = inputs[0]
+    updates = inputs[1]
+    dim = 0 if len(inputs) <= 2 else inputs[2].val
+    start = 0 if len(inputs) <= 3 else inputs[3]
+    end = x.shape[dim] if len(inputs) <= 4 else mb.minimum(x=inputs[4], y=x.shape[dim])
+    step = 1 if len(inputs) <= 5 else inputs[5]
+
+    assert dim is not None, "slice dim must be known at compile time"
+    assert 0 <= dim and dim < x.rank
+
+    # mb.torch_tensor_assign handles multi-dim slicing
+    # so we need to pad start, end, step from scalar to x.rank
+    starts = [0] * x.rank
+    starts[dim] = start
+    starts = mb.concat(values=starts, axis=0)
+    ends = list(x.shape)
+    ends[dim] = end
+    ends = mb.concat(values=ends, axis=0)
+    steps = [1] * x.rank
+    steps[dim] = step
+    steps = mb.concat(values=steps, axis=0)
+
+    # mb.torch_tensor_assign also have masks
+    begin_mask = [True] * x.rank
+    if start.val not in (0, -x.rank):
+        begin_mask[dim] = False
+    end_mask = [True] * x.rank
+    if end.val is None or end.val < x.shape[dim]:
+        end_mask[dim] = False
+    squeeze_mask = [False] * x.rank
+
+    updated_x = _translate_torch_tensor_assign(
+        x=x,
+        updates=updates,
+        begin=starts,
+        end=ends,
+        stride=steps,
+        begin_mask=begin_mask,
+        end_mask=end_mask,
+        squeeze_mask=squeeze_mask,
+        name=node.name,
+    )
+    context.add(updated_x)
+
+
+@register_torch_op
+def index_put(context, node):
+    inputs = _get_inputs(context, node, min_expected=3)
+    x = inputs[0]
     indices = inputs[1]
     values = inputs[2]
-    accumulate = inputs[3].val
-    rank = x.rank
+    accumulate = False if len(inputs) < 4 else inputs[3].val
     mode = "add" if accumulate else "update"
 
-    indices_type = indices[0].sym_type.get_primitive()
+    assert isinstance(indices, list), "indices must be a list of tensors"
+    # Usually indices is a list of non-None tensors, so we stack them and feed to mb.scatter_nd
+    # However, when there exists a whole slice, that index is represented as None
+    exist_slice = False
+    for index in indices:
+        if index is None:
+            exist_slice = True
+            break
+    if exist_slice:
+        # We have 2 ways to translate such torch.index_put, both have pros and cons
+        # 1. mb.scatter_nd
+        #    * pro: can handle accumulate or update
+        #    * con: can only have whole slice at the endmost dimensions
+        # 2. mb.torch_tensor_assign
+        #    * pro: can have whole slice at arbitrary dimension
+        #    * con: can only handle update
+        # Here we use mb.torch_tensor_assign
+        # TODO: explore how can we cover as many torch.index_put cases as possible
+        if accumulate:
+            raise NotImplementedError("If there exists whole slices, only update mode handled yet")
```

Comment:
I don't really understand this error message. So It's a safe bet that something looking at this with fresh eyes isn't going to understand it. Please add some context. 

Maybe something like: slices that cover an entire dimension is not supported.

Reply:
Polished error message
```
If there existed any whole slice (e.g. : in x[:, 0]), only torch.index_put(..., accumulate=False) handled yet
```